File: SDKs/amazonaws.com/appstream/2016-12-01/go/pkg/models/operations/deleteimage.go
package operations

import (
"openapi/pkg/models/shared")


type DeleteImageXAmzTargetEnum string

const (
    DeleteImageXAmzTargetEnumPhotonAdminProxyServiceDeleteImage DeleteImageXAmzTargetEnum = "PhotonAdminProxyService.DeleteImage"
)


type DeleteImageHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    XAmzTarget DeleteImageXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
    
}

type DeleteImageRequest struct {
    Headers DeleteImageHeaders 
    Request shared.DeleteImageRequest `request:"mediaType=application/json"`
    
}

type DeleteImageResponse struct {
    ConcurrentModificationException *interface{} 
    ContentType string 
    DeleteImageResult *shared.DeleteImageResult 
    OperationNotPermittedException *interface{} 
    ResourceInUseException *interface{} 
    ResourceNotFoundException *interface{} 
    StatusCode int64 
    
}

File: SDKs/amazonaws.com/appstream/2016-12-01/typescript/src/sdk/models/operations/deleteimage.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";

export enum DeleteImageXAmzTargetEnum {
    PhotonAdminProxyServiceDeleteImage = "PhotonAdminProxyService.DeleteImage"
}


export class DeleteImageHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: DeleteImageXAmzTargetEnum;
}


export class DeleteImageRequest extends SpeakeasyBase {
  @Metadata()
  headers: DeleteImageHeaders;

  @Metadata({ data: "request, media_type=application/json" })
  request: shared.DeleteImageRequest;
}


export class DeleteImageResponse extends SpeakeasyBase {
  @Metadata()
  concurrentModificationException?: any;

  @Metadata()
  contentType: string;

  @Metadata()
  deleteImageResult?: shared.DeleteImageResult;

  @Metadata()
  operationNotPermittedException?: any;

  @Metadata()
  resourceInUseException?: any;

  @Metadata()
  resourceNotFoundException?: any;

  @Metadata()
  statusCode: number;
}
